server/src/main/java/org/opensearch/search/aggregations/bucket/terms/StreamStringTermsAggregator.java

```diff
@@ -91,6 +93,13 @@ public InternalAggregation buildEmptyAggregation() {
 
     @Override
     public LeafBucketCollector getLeafCollector(LeafReaderContext ctx, LeafBucketCollector sub) throws IOException {
+        if (this.leafCollectorCreated) {
```

Comment:
we can use compareAndSet

Reply:
I've haven't thought it's under concurrent access.